sources/packages/api/src/services/archive-db/archive-db.service.ts

```diff
@@ -0,0 +1,60 @@
+import * as dayjs from "dayjs";
+import { Injectable } from "@nestjs/common";
+import { Student } from "src/database/entities";
+import { createConnection, Connection } from "typeorm";
+import { PostgresConnectionOptions } from "typeorm/driver/postgres/PostgresConnectionOptions";
+import { StudentLegacyData } from "../../types";
+
+const config: PostgresConnectionOptions = require("../../../ormconfig");
+
+@Injectable()
+export class ArchiveDbService {
+  private _connection?: Connection;
+
+  public get connection(): Connection {
+    return this._connection;
+  }
+
+  async init() {
+    if (this._connection) {
+      return;
+    }
+    try {
+      this._connection = await createConnection({
+        type: "postgres",
+        host: config.host,
+        port: config.port,
+        username: config.username,
+        password: config.password,
+        database: process.env.ARCHIVE_DB || "sfas_db",
+        logging: ["error", "info", "warn"],
+        migrations: ["./src/services/archive-db/migrations/*.ts"],
+        synchronize: false,
+        migrationsRun: false,
+      });
+    } catch (excp) {
+      console.error(`Unable to connect archive db for ${excp}`);
+      throw excp;
+    }
+  }
+
+  async query<T>(raw: string, parameters?: any[]): Promise<T | null> {
+    try {
+      await this.init();
+      return this._connection.query(raw, parameters);
+    } catch (excp) {
+      console.error(`Unable to query in archive db for ${excp}`);
+      return null;
+    }
+  }
+
+  async getIndividualPDStatus(
```

Comment:
Does it make sense to return an array here?
Would not be the case to return the unique record found or null?